### DISCUSSION TITLE
error: [CORRUPTION]Bad page: checksum mismatch

### DISCUSSION THREAD

**Discussion by MeiHsinChen** [apache/doris#41188] [A - General / Q&A]
Doris查詢表時，會出現checksum mismatch錯誤，確認該表的compaction狀態，發現增量壓縮有出現錯誤。
<img width="933" alt="image" src="https://github.com/user-attachments/assets/4d20b19b-21eb-41e3-885a-80dec7baa8e2">

手動執行增量壓縮也失敗。
除了重新建表，請問是否還有其他解決方式? 還有為什麼會出現這樣的問題呢?
謝謝!

手動執行增量壓縮:
curl -X POST "http://benode:port/api/compaction/run?tablet_id=15683&compact_type=cumulative"
執行失敗:
fail to do cumulative compaction. res=[CORRUPTION]Bad page: checksum mismatch (actual=3887400334 vs expect=757263053), file=/home/maychen/apache-doris/be/storage/data/687/15683/545958608/020000000000242d35437c2d23f12d8d0c9439f44bc078a8_0.dat

查詢表出現的錯誤訊息:
ERROR 1105 (HY000) at line 1: RuntimeException, msg: java.lang.RuntimeException: Failed to analyze following columns:[ss_addr_sk,ss_ticket_number,ss_wholesale_cost,ss_item_sk,ss_sold_date_sk,ss_ext_list_price,ss_store_sk] Reasons: org.apache.doris.common.UserException: errCode = 2, detailMessage = (127.0.0.1)[CANCELLED]Bad page: checksum mismatch (actual=3887400334 vs expect=757263053), file=/home/maychen/apache-doris/be/storage/data/687/15683/545958608/020000000000242d35437c2d23f12d8d0c9439f44bc078a8_0.dat,java.lang.RuntimeException:
 failed to read next block when merging rowsets of tablet 15683, error: [CORRUPTION]Bad page: checksum mismatch (actual=3887400334 vs expect=757263053), file=/home/maychen/apache-doris/be/storage/data/687/15683/545958608/020000000000242d35437c2d23f12d8d0c9439f44bc078a8_0.dat

**Comment by ixzc** [comment]
建议英文提问，如使用中文提问，可以去ask.selectdb.com中文技术论坛。

**Comment by ya-dao** [comment]
1. 检查涉及到的表select count(*)和select  * from xxx limit 1是否正常
2. 如果正常, 说明表tablet 有问题，内部修复了, 多副本的情况，如果有tablet出现问题，会自动设置成bad后，自动clone
3. 出现这个报错，可能是V2.0低版本的，建议能把版本拉到V2.1上
上述是技术人员给出的指导建议, 下面是我们排查到的be警告日志:
<img width="1896" alt="image" src="https://github.com/user-attachments/assets/d28bf225-08c5-42cf-88a9-765776945921">
